azure/converters/tags_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+func TestMapToTags(t *testing.T) {
+	tests := []struct {
+		name   string
+		srcMap map[string]*string
+		want   v1beta1.Tags
+	}{
+		{
+			name: "convert map to tags",
+			srcMap: map[string]*string{
```

Comment:
Although this is really simple, can we add a test for (1) the map is empty, and (2) what if the map value is a nil string pointer?